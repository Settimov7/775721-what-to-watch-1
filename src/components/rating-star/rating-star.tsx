import * as React from 'react';

interface Props {
  value: number;
  isChecked: boolean;
  onClick: React.MouseEventHandler<HTMLInputElement>;
}

export const RatingStar: React.FunctionComponent<Props> = (props) => {
  const {value, isChecked, onClick} = props;

  return (
    <React.Fragment>
      <input
        className="rating__input"
        id={`star-${value}`}
        type="radio"
        name="rating"
        value={value}
        checked={isChecked}
        onClick={onClick}
      />
      <label
        className="rating__label"
        htmlFor={`star-${value}`}
      >
        Rating {value}
      </label>
    </React.Fragment>
  );
};
