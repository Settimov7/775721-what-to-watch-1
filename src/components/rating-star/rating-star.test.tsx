import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {RatingStar} from './rating-star';

it(`Rating correctly renders`, () => {
  const onClick = jest.fn();
  const props = {
    value: 3,
    isChecked: true,
    onClick,
  };
  const ratingStar = renderer.create(
    <RatingStar {...props}/>
  ).toJSON();

  expect(ratingStar).toMatchSnapshot();
});
