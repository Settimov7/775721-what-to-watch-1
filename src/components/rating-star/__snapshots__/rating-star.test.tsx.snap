// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating correctly renders 1`] = `
Array [
  <input
    checked={true}
    className="rating__input"
    id="star-3"
    name="rating"
    onClick={[MockFunction]}
    type="radio"
    value={3}
  />,
  <label
    className="rating__label"
    htmlFor="star-3"
  >
    Rating 
    3
  </label>,
]
`;
